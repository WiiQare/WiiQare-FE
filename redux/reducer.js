import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  client: { register: {}, patient: {} },
};

export const ReducerSlice = createSlice({
  name: 'wiiqare',
  initialState,
  reducers: {
    setRegister: (state, action) => {
      state.client.register = { ...action.payload };
    },
    setPatientDispatch: (state, action) => {
      state.client.patient = { ...action.payload };
    },
  },
});

export const { setRegister, setPatientDispatch } = ReducerSlice.actions;

export default ReducerSlice.reducer;
