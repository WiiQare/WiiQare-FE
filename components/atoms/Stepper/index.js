import React, { useContext, useEffect } from "react";
import { FormContext } from "../../../pages/voucher/buy";

function Stepper() {

    const { activeStepIndex } = useContext(FormContext);

    useEffect(() => {
        const stepperItems = document.querySelectorAll(".stepper-item");
        stepperItems.forEach((step, i) => {
            if (i <= activeStepIndex) {
                step.classList.add("bg-primary", "text-white");

                if(i < activeStepIndex) step.nextSibling.classList.add('border-primary')

            } else {
                step.classList.remove("bg-primary", "text-white");
            }
        });
    }, [activeStepIndex]);

    return (
        <div className="w-full md:w-2/3 flex flex-row items-center justify-center py-4">
            <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
                1
            </div>
            <div className="flex-auto border-t-2"></div>
            <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
                2
            </div>
            <div className="flex-auto border-t-2"></div>
            <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
                3
            </div>
        </div>
    );
}

export default Stepper;
