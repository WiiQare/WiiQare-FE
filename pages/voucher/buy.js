import { createContext, useState } from 'react';
import Head from 'next/head';
import { useRouter } from 'next/router';
import DashboardLayout from '../../layouts/Dashboard';
import CardHeader from '../../components/atoms/Card/Header';
import Stepper from '../../components/atoms/Stepper';
import Step from '../../components/atoms/Stepper/step';
export const FormContext = createContext();

const Page = () => {
  const { step, redirect_status, payment_intent } = useRouter().query;
  const [activeStepIndex, setActiveStepIndex] = useState(
    step == 'end' && redirect_status == 'succeeded' ? 2 : 0,
  );
  const [formData, setFormData] = useState({});

  return (
    <>
      <Head>
        <title>Envoyer un pass santé</title>
      </Head>
      <div className="p-2 space-y-6 md:py-8 md:px-6">
        <CardHeader
          title={'Envoyer un pass santé'}
          breadcrumbs={[
            {
              item: 'Accueil',
              link: '/',
            },
            {
              item: 'Envoyer un pass santé',
              link: '/voucher/buy',
            },
          ]}
          download={false}
          print={false}
        />

        <section className="w-full pb-20 md:pb-0">
          <div className="w-full overflow-hidden md:col-span-2 rounded-lg py-8 flex flex-col gap-6 bg-white drop-shadow-sm">
            <div className="px-8 space-y-8">
              <div>
                <h2 className="font-semibold text-3xl">
                  Envoyer un pass santé
                </h2>
              </div>
              <div className="flex flex-col gap-4">
                <FormContext.Provider
                  value={{
                    activeStepIndex,
                    setActiveStepIndex,
                    formData,
                    setFormData,
                    payment_intent,
                  }}
                >
                  <div className="flex flex-col items-center justify-start">
                    <Stepper />
                    <Step />
                  </div>
                </FormContext.Provider>
              </div>
            </div>
          </div>
        </section>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;
