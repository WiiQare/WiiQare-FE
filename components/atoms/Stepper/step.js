import React, { useContext } from "react";
import { FormContext } from "../../../pages/voucher/buy";
import Identity from "./Forms/identity";
import Payment from "./Forms/payment";
import Send from "./Forms/send";
import Identity2 from "./Forms/identity2";
import Payment2 from "./Forms/payment2";


function Step({data}) {
  const { activeStepIndex } = useContext(FormContext);
  let stepContent;

  switch (activeStepIndex) {
    case 0:
      stepContent = <Identity2 />;
      break;

    case 1:
      stepContent = <Payment2 data={data} />;
      break;
    case 2:
      stepContent = <Send />;
      break;
    default:
      break;
  }

  return stepContent;
}

export default Step;
