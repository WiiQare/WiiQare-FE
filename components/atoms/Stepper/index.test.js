import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Stepper from '.';
import { FormContext } from '../../../pages/voucher/buy';
import { SessionProvider } from 'next-auth/react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import { QueryClientProvider, QueryClient } from 'react-query';

describe('Identity2', () => {
  let component;
  beforeEach(() => {
    const queryClient = new QueryClient();
    const res = render(
      <SessionProvider session={{ user: { data: { userId: 'random123' } } }}>
        <QueryClientProvider client={queryClient}>
          <Provider store={store}>
            <FormContext.Provider value={{ activeStepIndex: 0 }}>
              <Stepper />
            </FormContext.Provider>
          </Provider>
        </QueryClientProvider>
      </SessionProvider>,
    );
    component = res.container;
  });

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
