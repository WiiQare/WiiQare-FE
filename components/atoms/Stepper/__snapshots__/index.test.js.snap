// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identity2 should render the component 1`] = `
<div>
  <div
    class="w-full md:w-2/3 flex flex-row items-center justify-center py-4 mb-14 md:mb-16"
  >
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full bg-primary text-white"
      >
        1
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Identité 
        <span
          class="hidden md:inline"
        >
          du beneficiare
        </span>
         
      </span>
    </div>
    <div
      class="flex-auto border-t-2"
    />
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full"
      >
        2
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Résumé 
        <span
          class="hidden md:inline"
        >
          de paiement
        </span>
         
      </span>
    </div>
    <div
      class="flex-auto border-t-2"
    />
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full"
      >
        3
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Généré un 
        <span
          class="hidden md:inline"
        >
          Pass Santé 
        </span>
         
      </span>
    </div>
  </div>
</div>
`;
